src/inbox/mod_inbox_utils.erl

```diff
@@ -139,35 +144,37 @@ has_chat_marker(Packet, Markers) ->
             lists:member(Marker, Markers)
     end.
 
--spec maybe_write_to_inbox(Host, User, Remote, Packet, WriteF) -> ok | {ok, integer()} when
+-spec maybe_write_to_inbox(Host, User, Remote, Packet, TS, WriteF) -> ok | {ok, integer()} when
       Host :: host(),
       User :: jid:jid(),
       Remote :: jid:jid(),
       Packet :: exml:element(),
-      %% WriteF is write_to_receiver_inbox/4 or write_to_sender_inbox/4
+      TS :: integer(),
+      %% WriteF is write_to_receiver_inbox/5 or write_to_sender_inbox/5
       WriteF :: fun().
-maybe_write_to_inbox(Host, User, Remote, Packet, WriteF) ->
+maybe_write_to_inbox(Host, User, Remote, Packet, TS, WriteF) ->
     case mod_inbox_utils:has_chat_marker(Packet) of
         true ->
             ok;
         false ->
-            FromBin = jid:to_binary(User),
-            Packet2 = mod_inbox_utils:fill_from_attr(Packet, FromBin),
-            WriteF(Host, User, Remote, Packet2)
+            Packet2 = mod_inbox_utils:fill_from_attr(Packet, User),
```

Comment:
Module is redundant for a local call

Reply:
Like a bunch of others in this module 👍🏽